homeassistant/components/repetier/sensor.py

```diff
@@ -0,0 +1,294 @@
+"""Support for monitoring Repetier Server Sensors."""
+import logging
+from datetime import timedelta, datetime
+
+from homeassistant.components.repetier import (SENSOR_TYPES)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['repetier']
+
+SCAN_INTERVAL = timedelta(seconds=5)
+TIME_FORMAT = '%Y-%m-%d %H:%M:%S'
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the available Repetier Server sensors."""
+    import pyrepetier
+    if discovery_info is None:
+        return
+
+    id = discovery_info['printer']
+    url = discovery_info['url']
+    port = discovery_info['port']
+    apikey = discovery_info['apikey']
+    monitored_conditions = discovery_info['sensors']
+
+    def refresh(event_time):
+        """Update sensors."""
+        _addDevice()
+
+    track_time_interval(hass, refresh, SCAN_INTERVAL)
+
+    def _addDevice():
+        """Add devices."""
+        server = pyrepetier.Repetier(
+            url=url,
+            port=port,
+            apikey=apikey)
+        printers = server.getPrinters()
+
+        devices = []
+
+        for sens_type in monitored_conditions:
+            if sens_type == 'Temperatures':
+                nozzles = server.Nozzle(id)
+                if nozzles is not None:
```

Comment:
But if nozzles is not none we should proceed, so I don't see how you would invert this?!

Reply:
```py
if nozzles is None:
    continue
...
```

Reply:
Will test

Reply:
So you mean this?
`                nozzles = server.Nozzle(id)
                if nozzles is not None:
                    for nozzle in nozzles:
                        state = hass.states.get('sensor.' +
                                                printers[id]['slug'] +
                                                '_nozzle_' + str(nozzle))
                        if not state:
                            svar = {'nozzle': nozzle,
                                    'name': printers[id]['slug'],
                                    'id': prn_id,
                                    'server': {'url': url,
                                               'port': port,
                                               'apikey': apikey},
                                    'sens_type': sens_type}
                            new_sensor = RepetierSensor('nozzle', svar)
                            devices.append(new_sensor)
`
I have been looking at the code this evening, but can't wrap my head around this (perhaps it's because Python is still pretty new to me), but wouldn't your example just skip out of the parent if-block? That would be most unfortunate.

Reply:
Yes it would skip out of the if block. I missed that indentation.

This comment isn't relevant at this place anymore since we should refactor and only send the correct sensor type in the discovery_info when loading the platform.